### PR TITLE
#367: Include parent projects in property resolution

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Includes parent projects in property resolution for property updates. Instead of only considering the model of the current project, will also consult parent models to see if there are property definitions for properties defined in the current project.

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
@slawekjaranowski Please review.

File: src/main/java/org/codehaus/mojo/versions/api/PomHelper.java [hunk 1]
```diff
@@ -103,15 +105,12 @@ public static Model getRawModel( MavenProject project )
      * @throws IOException if the file is not found or if the file does not parse.
      */
     public static Model getRawModel( File moduleProjectFile )
-        throws IOException
+            throws IOException
     {
-        try ( FileInputStream input = new FileInputStream( moduleProjectFile ) )
+        try ( Reader reader = new BufferedReader( new InputStreamReader( Files.newInputStream( moduleProjectFile
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/mojo/versions/api/PomHelper.java:110]: There is `Files.newBufferedReader`

**Comment by ajarmoniuk** [comment]
As you can see, tried that, but it caused the `it-encoding-001` test to fail. `Files.newBufferedReader` assumes `UTF-8` for input, which is not always the case.

**Comment by madorb** [comment]
well this was an unwanted surprise! is there any way to disable this functionality? 

consider users of say, spring boot that have [spring-boot-dependencies](https://search.maven.org/artifact/org.springframework.boot/spring-boot-dependencies/2.7.1/pom) as their parent

it's _quite_ unhelpful to now see the full list of all properties from the parent by default. Most of those are for dependencies any given application doesn't use. 

This radical of a change really should come with a configuration option either in the xml or command line.

**Comment by ajarmoniuk** [comment]
@madorb yes, but only in the upcoming release, see #795

**Pull request merged**